qabelbox/src/main/java/de/qabel/qabelbox/activities/MainActivity.java

```diff
@@ -508,30 +515,79 @@ public void onClick(DialogInterface dialog, int which) {
                                     case R.id.share:
                                         ExternalApps.share(self, getUri(boxObject), getMimeType(boxObject));
                                         break;
+                                    case R.id.shareto_qabeluser:
+                                        shareToQabelUser(boxObject);
+                                        break;
                                     case R.id.delete:
                                         delete(boxObject);
                                         break;
+                                    case R.id.unshare:
+                                        filesFragment.getBoxNavigation().removeFileMetadata((BoxFile) boxObject);
+                                        break;
                                     case R.id.export:
-                                        // Export handled in the MainActivity
-                                        if (boxObject instanceof BoxFolder) {
-                                            Toast.makeText(self, R.string.folder_export_not_implemented,
-                                                    Toast.LENGTH_SHORT).show();
-                                        } else {
-                                            onExport(filesFragment.getBoxNavigation(), boxObject);
-                                        }
+                                        onExport(filesFragment.getBoxNavigation(), boxObject);
                                         break;
                                 }
                             }
-                        }).show();
+                        });
+
+                filterSheet(boxObject, sheet);
+                sheet.show();
             }
         });
     }
 
+    protected void filterSheet(BoxObject boxObject, BottomSheet.Builder sheet) {
```

Comment:
I replaced this instanceof construct and the call to the deprecated sheet.remove with an other concept in https://github.com/Qabel/qabel-android/pull/260. This should not stop this PR from merging and could be merged afterwards.